scripts/compress_images.py

```
@@ -0,0 +1,233 @@
+        """
+        self.input_path = pathlib.Path(input_path)
+        self.output_dir = output_dir
+        self.tolerance = tolerance
+        self.logger = logging.getLogger(__name__)
+
+    def get_compression_type(self, file_extension: str) -> str:
+        """Determine compression type based on file extension.
+
+        Args:
+            file_extension: str. The file extension of the image.
+
+        Returns:
+            type: str. Compression type (Zip or LZW).
+        """
+        return (
+            'Zip' if file_extension in 
+            self.IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION
+            else 'LZW'
+        )
+
+    def find_compressible_images(self) -> List[CompressedImageInfo]:
+        """Find images that can be compressed further.
+
+        Returns:
+            result_images: List[]. List of compressible images.
+        """
+        result_images: List[CompressedImageInfo] = []
+
+        for file_path in self.input_path.glob('**/*.*'):
+            file_extension = file_path.suffix.lower()
+            if file_extension not in self.ALL_IMAGES_EXTENSIONS:
+                continue
+
+            with Image.open(file_path):
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    temp_compressed = (
+                        pathlib.Path(tmpdir) / f'compressed_{file_path.name}'
+                    )
+
+                    compression_type = self.get_compression_type(file_extension)
+
+                    cmd = [
+                        'gm', 'convert',
+                        str(file_path),
+                        '-strip',
+                        '-compress', compression_type,
+                        str(temp_compressed)
+                    ]
+
+                    result = subprocess.run(
+                        cmd, capture_output=True, text=True, check=False
+                    )
+                    if result.returncode == 0 and temp_compressed.exists():
```

are there situations where we expect that this should happen? If not, we should throw an error (could just set `check=True` in `subprocess.run`

---

Gud point. Done.